src/mem3/src/mem3_shards.erl

```diff
@@ -320,8 +360,20 @@ load_shards_from_disk(DbName) when is_binary(DbName) ->
 load_shards_from_db(#db{} = ShardDb, DbName) ->
     case couch_db:open_doc(ShardDb, DbName, [ejson_body]) of
     {ok, #doc{body = {Props}}} ->
+        Seq = couch_db:get_update_seq(ShardDb),
         Shards = mem3_util:build_ordered_shards(DbName, Props),
-        gen_server:cast(?MODULE, {cache_insert, DbName, Shards}),
+        case maybe_spawn_shard_writer(DbName, Shards) of
```

Comment:
This doesn't seem right. We link writer to the caller. If caller process dies it take writer with it and writer never gets a chance to clean itself up thus blocking any other future writers from being created because of 

```
case ets:member(?OPENERS, DbName) of   true ->   ignore;
```